NotifierTemplate/Model/TemplateGetter/FilesystemTemplateGetter/GetTemplateFile.php

```diff
@@ -69,19 +92,39 @@ public function execute(string $adapterCode, string $templateId): string
             FileSystem::normalizePath($templateFile)
         );
 
-        $templatePath = $this->reader->getModuleDir('', $module). '/' .
-            FilesystemTemplateRepository::TEMPLATE_MODULE_DIR;
-
         if ($adapterCode) {
-            $fullPath = $templatePath . '/' . $adapterCode . '/' . $filePath;
+            $moduleName = $this->getModuleNameByAdapterCode($adapterCode);
+
+            $templatePath = $this->reader->getModuleDir('', $moduleName) . '/' .
+                FilesystemTemplateRepository::TEMPLATE_MODULE_DIR;
         } else {
-            $fullPath = $templatePath . '/' . $filePath;
+            $templatePath = $this->reader->getModuleDir('', $module) . '/' .
+                FilesystemTemplateRepository::TEMPLATE_MODULE_DIR;
         }
 
+        $fullPath = $templatePath . '/' . $filePath;
+
         if (!$this->file->fileExists($fullPath)) {
             throw new FileSystemException(__('Template %1 does not exist', $adapterCode . '/' . $templateId));
         }
 
         return $fullPath;
     }
+
+    /**
+     * Get Module Name of Adapter by Adapter Code
+     *
+     * @param string $adapterCode
+     * @return string
+     * @throws \Magento\Framework\Exception\NoSuchEntityException
+     * @throws \ReflectionException
+     */
+    protected function getModuleNameByAdapterCode(string $adapterCode): string
+    {
+        $adapterClass = $this->adapterRepository->getAdapterByCode($adapterCode);
+        $adapterEngine = $adapterClass->getEngine();
+        $reflectionClass = new \ReflectionClass(get_class($adapterEngine));
```

Comment:
Mapping should be provided via DI

Reply:
Updated,
added mapping for template file to di.xml